packages/analytics/src/helpers.ts

```diff
@@ -95,33 +57,86 @@ export function getOrCreateDataLayer(dataLayerName: string): DataLayer {
  */
 function wrapGtag(
   gtagCore: Gtag,
-  initializedIdPromisesMap: { [gaId: string]: Promise<void> }
+  initializationPromisesMap: { [appId: string]: Promise<string> },
+  dynamicConfigPromisesList: Array<Promise<DynamicConfig>>,
+  measurementIdToAppId: { [measurementId: string]: string }
 ): Function {
-  return (
-    command: 'config' | 'set' | 'event',
-    idOrNameOrParams: string | ControlParams,
+  /**
+   * Wrapped gtag logic when gtag is called with 'config' command.
+   *
+   * @param measurementId GA Measurement ID.
+   * @param gtagParams Config params.
+   */
+  async function gtagOnConfig(
```

Comment:
Not a big fan of nested functions. They provides some level of code separation, but `wrapGtag()` is still big. Can you move them to the top level?

Reply:
Done.